stl/inc/random

```diff
@@ -2932,15 +2932,32 @@ private:
             _Ty _Vx1;
             _Ty _Vx2;
             _Ty _Sx;
-            for (;;) { // reject bad values
+            for (;;) { // reject bad values to avoid generating NaN/Inf on the next calculations
                 _Vx1 = 2 * _NRAND(_Eng, _Ty) - 1;
                 _Vx2 = 2 * _NRAND(_Eng, _Ty) - 1;
                 _Sx  = _Vx1 * _Vx1 + _Vx2 * _Vx2;
-                if (_Sx < 1) {
+                if (_Sx < _Ty{1} && _Vx1 != _Ty{0} && _Vx2 != _Ty{0}) {
+                    // good values!
                     break;
                 }
             }
-            const auto _Fx = static_cast<_Ty>(_CSTD sqrt(-2.0 * _CSTD log(_Sx) / _Sx));
+
+            _Ty _LogSx;
+            if (_Sx > 1.0e-4) {

```

Comment:
This is comparing against a `double`, so it'll widen `float` to `double`. Should it instead be comparing against `_Ty{1e-4}`, like the comparison against `_Ty{1e-12}` below? Note, however, that there is a value difference. I am uncertain what to do here (I suspect that the _precise_ threshold isn't terribly important so using `_Ty` should be fine).

```
C:\Temp>type meow.cpp
#include <stdio.h>

int main() {
    printf("%.1000g\n", 1e-4);
    printf("%.1000g\n", 1e-4f);
}

C:\Temp>cl /EHsc /nologo /W4 meow.cpp && meow
meow.cpp
0.000100000000000000004792173602385929598312941379845142364501953125
9.99999974737875163555145263671875e-05
```

Reply:
I suggested this, so I'll confirm that it's not intended to be a precise threshold. The precise one would be based on preventing underflow in ` _Vx1 * _Vx1 + _Vx2 * _Vx2` and overflow in `2*log(_Sx)/_Sx`. For `float` those limits are about 4e-16 and 5e-37, so 1e-4 is very conservative. `_Ty{1e-4}` would be better, to avoid the unnecessary conversion cost.

Reply:
Thanks - I'll go ahead and push changes then.